otherlibs/stdune/src/copyfile_stubs.c

```diff
@@ -0,0 +1,42 @@
+#include <caml/memory.h>
+#include <caml/mlvalues.h>
+
+#if defined(__APPLE__)
+#include <caml/threads.h>
+#include <caml/unixsupport.h>
+#include <copyfile.h>
+
+CAMLprim value stdune_copyfile(value v_from, value v_to) {
+  CAMLparam2(v_from, v_to);
+  copyfile_state_t state = copyfile_state_alloc();
+  const char *from = caml_stat_strdup(v_from);
+  const char *to = caml_stat_strdup(v_to);
+  caml_release_runtime_system();
+  int ret = copyfile(from, to, state, COPYFILE_CLONE);
+  copyfile_state_free(state);
+  caml_stat_free(from);
+  caml_stat_free(to);
+  caml_acquire_runtime_system();
```

Comment:
I would invert the calls here; touching runtime data structures without holding the runtime lock makes me nervous:
```suggestion
  caml_acquire_runtime_system();
  caml_stat_free(from);
  caml_stat_free(to);
```

Reply:
I actually followed the manual here:

```
  /* Free the copy of the string, which we might as well do before
     acquiring the runtime system to benefit from parallelism. */
  caml_stat_free(name);
  /* Re-acquire the OCaml run-time system */
  caml_acquire_runtime_system();
```

https://v2.ocaml.org/manual/intfc.html

Reply:
Right, thanks for the pointer, I learned something new.